### ISSUE TITLE
uPlot charting lib

### ISSUE THREAD

**Issue by leeoniya** [home-assistant/frontend#5587]
## The request

Hi guys, this is a bit of a continuation of https://github.com/leeoniya/uPlot/issues/181 opened by @Misiu. I did a quick search for [graph/chart related issues](https://github.com/home-assistant/frontend/search?o=desc&q=chart+OR+plot+OR+graph&s=updated&type=Issues) here and have found that quite a few could potentially be resolved by uPlot (e.g. #3332, #4187, #4421, #5362, #5451... maybe: #2061, #5231, #3909..). i'm a fan of home-assistant and if you guys feel like this is a worthy endeavor to explore, i'd be happy to provide some assistance.

## The alternatives

fixing the bugs with the existing charting lib, or switching to another one.

## Additional information

cc @balloob @bramkragten 

thanks!

**Comment by balloob** [comment]
Yeah, our charts haven't been updated since eh, forever really. (nor has been our history panel for that matter). I would definitely be open to replacing it with anything better that would .

I personally don't do too much work on the frontend anymore, and I know that Bram is busy too. So although we're open for it, it would not be something we would work on right now. However contributions are welcome.

**Comment by leeoniya** [comment]
do you guys have any mock dashboards that can be hooked up to a mock server that can generate fake but realistic data. if something like this is available then i can probably prototype something, but i definitely do not have the spare cycles to make a realistic home-assistant deployment to get this off the ground.

is it possible to clone & build this `frontend` repo and config the UI to use an external demo server like https://demo.home-assistant.io/ ?

**Comment by bramkragten** [comment]
The demo is not a server, it is all mocked in the application (https://github.com/home-assistant/frontend/tree/dev/demo) itself, so you can run that without problems in you dev env. Besides that, we have a `demo` backend integration that provides fake integration data (although that might not be easy for history).

**Comment by balloob** [comment]
This will get you running in the demo, which contains fake data:

```
git clone https://github.com/home-assistant/frontend.git
cd frontend
script/bootstrap
./demo/script/develop_demo
```

**Comment by Misiu** [comment]
@leeoniya I can get the data from last week from my sensors. I don't have them that much but You will get the data that is coming directly from the server.
I'll create a Gist and link it here.
Is that OK?

**Comment by leeoniya** [comment]
> This will get you running in the demo, which contains fake data:

thanks! i tried this, but i'm on windows and it's not that simple to get things working. i can boot into Manjaro and probably get this working but...

the sheer amount of dependencies [1] that must be installed is simply eye-watering to me D:, when typically my dependencies count is near 0 and dev-deps is just rollup + terser. the date-picker here alone is enormous (maybe due to polymer?), and i've written a high-performance date picker myself that's < 100 LOC [2] for the essential functionality.

i'm afraid i'll have to tap out on going whole-hog here. but if someone has a setup that already works, i'll glady answer any uPlot-related questions and try to close out the graph-related issues in this repo. uPlot's API is minimal, and hopefully very straightforward.

[1] https://github.com/home-assistant/frontend/blob/dev/package.json#L26-L193
[2] https://domvm.github.io/domvm/demos/playground/#calendar

**Comment by balloob** [comment]
This is an application, not a lib 🤷
